pkg/executor/push_test.go

```diff
@@ -57,21 +57,24 @@ func TestDockerConfLocation(t *testing.T) {
 	if unset != unsetExpected {
 		t.Errorf("Unexpected default Docker configuration file location: expected:'%s' got:'%s'", unsetExpected, unset)
 	}
-
-	if err := os.Setenv(dcfg, "/kaniko/.docker"); err != nil {
+	tmpDir, _ := ioutil.TempDir("", "*")
+	dir := filepath.Join(tmpDir, "/kaniko/.docker")
+	os.MkdirAll(dir, os.ModePerm)
+	if err := os.Setenv(dcfg, dir); err != nil {
 		t.Fatalf("Failed to set DOCKER_CONFIG: %v", err)
 	}
 	kanikoDefault := DockerConfLocation()
-	kanikoDefaultExpected := "/kaniko/.docker/config.json" // will fail on Windows
+	kanikoDefaultExpected := filepath.Join(tmpDir, "/kaniko/.docker/config.json") // will fail on Windows
 	if kanikoDefault != kanikoDefaultExpected {
 		t.Errorf("Unexpected kaniko default Docker conf file location: expected:'%s' got:'%s'", kanikoDefaultExpected, kanikoDefault)
 	}
 
-	if err := os.Setenv(dcfg, "/a/different/path"); err != nil {
+	differentPath, _ := ioutil.TempDir("", "differentPath")
```

Comment:
please add a defer to clean up the. temp directory.
Can you please fail Fatal if `ioutil.TempDir` sees and error instead of ignoring it.